File: Sorting-Algorithms/BubbleSort/gelvis/gelvis.py
def bubble_sort(array):
    unsorted_until_array_length = len(array) - 1
    is_sorted = False
    while not is_sorted:
        is_sorted = True
        for i in range(unsorted_until_array_length):
            if array[i] > array[i+1]:
                is_sorted = False
                array[i] , array[i+1] = array[i+1], array[i]
        unsorted_until_array_length = unsorted_until_array_length - 1

list = [8,90,78,26,30,2]
bubble_sort(list)

print(list)

File: Sorting-Algorithms/BubbleSort/gelvis.cpp
#include <iostream>

void bubble_sort(int array[], int length) {
    int unsorted_until = length - 1;
    bool isSorted = false;

    while(!isSorted){
        isSorted = true;

        for(int i = 0; i< unsorted_until; i++) {
            if(array[i] > array[i+1]) {
                isSorted = false;
                int temp = array[i];
                array[i] = array[i+1];
                array[i+1] = temp;
            }
        }
        unsorted_until = unsorted_until - 1;
    }

}

int main() {
    int list[] = {13, 45, 3, 89, 2, 57, 3};
    int length = sizeof(list) / sizeof(list[0]);
    bubble_sort(list, length);

    for (int i = 0; i < length; i++) {
        std::cout << list[i] << " ";
    }

    return 0;
}